react/features/invite/components/add-people-dialog/web/AddPeopleDialog.js

```diff
@@ -1,478 +1,209 @@
-                    </div>
-                    { translateToHTML(t, 'addPeople.footerText') }
-                </div>
-            };
-        }
-
-        return (
-            <Dialog
-                okDisabled = { this._isAddDisabled() }
-                okKey = 'addPeople.add'
-                onSubmit = { this._onSubmit }
-                titleKey = 'addPeople.title'
-                width = 'medium'>
-                <div className = 'add-people-form-wrap'>
-                    { this._renderErrorMessage() }
-                    <MultiSelectAutocomplete
-                        footer = { footerText }
-                        isDisabled = { isMultiSelectDisabled }
-                        loadingMessage = { t(loadingMessage) }
-                        noMatchesFound = { t(noMatches) }
-                        onItemSelected = { this._onItemSelected }
-                        onSelectionChange = { this._onSelectionChange }
-                        placeholder = { t(placeholder) }
-                        ref = { this._setMultiSelectElement }
-                        resourceClient = { this._resourceClient }
-                        shouldFitContainer = { true }
-                        shouldFocus = { true } />
-                </div>
-            </Dialog>
-        );
-    }
-
-    _invite: Array<Object> => Promise<*>
-
-    _isAddDisabled: () => boolean;
-
-    _onItemSelected: (Object) => Object;
+    t: Function,
 
     /**
-     * Callback invoked when a selection has been made but before it has been
-     * set as selected.
-     *
-     * @param {Object} item - The item that has just been selected.
-     * @private
-     * @returns {Object} The item to display as selected in the input.
+     * Method to update the dial in numbers.
      */
-    _onItemSelected(item) {
-        if (item.item.type === 'phone') {
-            item.content = item.item.number;
-        }
-
-        return item;
-    }
+    updateNumbers: Function
+};
 
-    _onSelectionChange: (Map<*, *>) => void;
+/**
+ * Invite More component.
+ *
+ * @returns {React$Element<any>}
+ */
+function AddPeopleDialog({
+    conferenceName,
+    dialIn,
+    hideInviteContacts,
+    inviteUrl,
+    liveStreamViewURL,
+    localParticipantName,
+    locationUrl,
+    t,
+    updateNumbers }: Props) {
+    const [ phoneNumber, setPhoneNumber ] = useState(undefined);
+
+    /**
+     * Updates the dial-in numbers.
+     */
+    useEffect(() => {
+        if (!dialIn.numbers) {
+            updateNumbers();
+        }
+    }, []);
 
     /**
-     * Handles a selection change.
+     * Sends analytics events when the dialog opens/closes.
      *
-     * @param {Map} selectedItems - The list of selected items.
-     * @private
      * @returns {void}
      */
-    _onSelectionChange(selectedItems) {
-        this.setState({
-            inviteItems: selectedItems
-        });
-    }
+    useEffect(() => {
+        sendAnalytics(createInviteDialogEvent(
```

Comment:
This wouldd be triggered on every render, we probably don't want to do that

Reply:
To my knowledge, by passing the `[]` as a second arg to the effect, it only runs it once per component lifecycle, acting as a sort of `componentDidMount` and `componentWillUnmount` in one (via the returned cleanup function).